Vkotha master

vivekkothakota opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

vivekkothakota committed:
Update Azure Migrate Swagger file with newly added API

Update Azure Migrate Swagger file with newly added API's and respective examples.

vivekkothakota committed:
Updated the response schema for Assessment Options API

Updated the response schema for Assessment Options API

vivekkothakota committed:
Added the schema for Get projects in subscription API

Added the schema for Get projects in subscription API

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-node

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-go

vivekkothakota committed:
Fixing the build failure.

Fixing the build failure.

vivekkothakota committed:
Fixed issues found while validating examples with swagger

Fixed issues found while validating examples with swagger

vivekkothakota committed:
FIxing a build failure

vivekkothakota commented:
Ping....
Are you waiting on me? is there something blocked from my side?

annatisch commented on specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json:
```diff
@@ -1670,9 +1709,183 @@
+      "get": {
+        "summary": "Get all projects.",
+        "description": "Get all the projects in thh subscription.",
+        "operationId": "ProjectsInSubscription_List",
```
This operation should be called `Projects_List` and the operation that lists according to resource group should be called something like `Projects_ListByResourceGroup` - would this change be possible or do you already have published SDKs that will break?
Otherwise, this would be changed to `Projects_ListBySubscription`.

Additionally there's a typo in the description: `thh`.

vivekkothakota replied:
Done.

annatisch commented on specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json:
```diff
@@ -1670,9 +1709,183 @@
+            "description": "Service Unavailable."
+          }
+        },
+        "x-ms-pageable": {
```
I don't think you can make this operation pageable - the list response has two separate list parameters.
When I try to generate a Python SDK it fails in attempting to make this operation pageable.

vivekkothakota replied:
Is there a way, we can suppress this warning for now? We will make the response pageable in the next API version change.

annatisch replied:
Hi @vivekkothakota - this is currently an error, so it will need to be modified. When you say it will be supported in a subsequent API version, do you mean that the response will have a `nextLink`? Or do you mean that the model will be restructured to have a single iterable array?

The main reason it is failing to generate is because you need to specify the name of the array you wish to be iterated in the paged response. The response model currently defines two different arrays.
Some languages do not support multiple arrays in the response - so whichever array you select to be the iterating array, the other array will be lost during deserliazation. If you still want to do this, you need to define `itemName` as described the [x-ms-pageable docs](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-pageable).

Alternatively, simply remove x-ms-ageable for now, and both arrays in the response will be deserilized as a normal model.

vivekkothakota replied:
thank you. remove x-ms-pageable for now..

annatisch commented on specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json:
```diff
@@ -1670,9 +1709,183 @@
+      "delete": {
+        "summary": "Delete all projects in the subscription.",
+        "description": "Delete all projects in the subscription. Deleting non-existent project is a no-operation.",
+        "operationId": "SubscriptionNotification_Delete",
```
Given the description of this operation - I would probably change the operation ID to something like "Projects_DeleteBySubscription" or "Projects_DeleteAll".
But you better understand the chosen naming convention, so up to you :)

vivekkothakota replied:
Done.

annatisch commented:
Hi @vivekkothakota - sorry about the late review - I requested just a couple of minor changes and we're good to go :)

vivekkothakota committed:
Resolved review comments

Resolved review comments

vivekkothakota committed:
Removing x-ms-pagable property for now.

Removing x-ms-pagable property for now.

annatisch commented:
@vivekkothakota - This PR is missing ARM sign-off. Since many days have passed, please let me know if you prefer to: 
1) follow up with ravbhatnagar and wait, or 
2) I can merge the PR without ARM’s team sign-off. Please note that even after merge, you should follow up with ARM team regarding ARM issues and fixes may impact the specs and/or generated SDKs.

ping @ravbhatnagar 

Thanks!

vivekkothakota commented:
Thank you @annatisch. I did send an email to @ravbhatnagar, lets wait for a day for his response before merging.

vivekkothakota committed:
Removed all the error cases and add a "default" response which captures all error cases.

Removed all the error cases and add a "default" response which captures all error cases.

vivekkothakota commented on specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json:
```diff
@@ -1670,14 +1709,188 @@
     }
   },
   "paths": {
+    "/subscriptions/{subscriptionId}/deleteSubscriptionNotification": {
```
Removed.

vivekkothakota commented on specification/migrate/resource-manager/Microsoft.Migrate/stable/2018-02-02/migrate.json:
```diff
@@ -1670,14 +1709,188 @@
+              "$ref": "#/definitions/ProjectResultList"
+            }
+          },
+          "401": {
```
Done.

annatisch commented:
Thanks @ravbhatnagar - looks like we're good to go? :)

ravbhatnagar commented:
Looks good!
